tt_metal/impl/dispatch/lock_free_queue.hpp

```diff
@@ -43,8 +43,8 @@ class LockFreeQueue {
 
     public:
         // Optional - Set these if the worker and parent thread state needs to be tracked
-        std::atomic<uint64_t> worker_thread_id = 0;
-        std::atomic<uint64_t> parent_thread_id = 0;
+        std::atomic<std::thread::id> worker_thread_id;
```

Comment:
It seems like we might not be initializing these attributes correctly in the `LockFreeQueue()` constructor now that you've removed the defaults?

As an aside.. is there any reason why we can't move this under private?

Reply:
Hey Joseph, based on the logic around how these values are being used I think its okay if we remove the defaults (cpp doesn't allow assigning a uint32_t to a thread_id). The reason is that we compare thread ids to these values if async mode is enabled (and if it is, we will initialize these values to the correct thread ids).

Currently not private because the parent object (`WorkExecutor`) needs to access them. I can clean this up in a follow on PR, if that't okay.

Reply:
okay, please do